### ISSUE TITLE
Question regarding IMU plugin

### ISSUE THREAD

**Issue by JaFeKl** [tu-darmstadt-ros-pkg/hector_gazebo#51]
Hello guys, 

I recently took a closer look into your IMU plugin and I'm a little bit confused by the following lines of code:

`rate = this->offset_.rot.GetInverse()
           * (2.0 * acos(std::max(std::min(delta.w, 1.0), -1.0)) * math::Vector3(delta.x, delta.y, delta.z).Normalize() / dt);`

shouldn't it be 
`rate = rot.GetInverse()
           * (2.0 * acos(std::max(std::min(delta.w, 1.0), -1.0)) * math::Vector3(delta.x, delta.y, delta.z).Normalize() / dt);` ?

, since you want to rotate the calculated angular rate from the world into the body reference frame and not only rotate it by the "rpyOffset" value.

I would really appreciate your comments on that, maybe it's just an error in reasoning by myself :)

**Comment by JaFeKl** [comment]
or where else is the angular velocity rotated from world into the body frame?
for the acceleartion it's clear to me by using `rot.RotateVectorReverse()` but  I can't find any similiar term regarding the angular velocity...

**Comment by meyerj** [comment]
The body angular rate is derived from the two orientation quaternions at the current and the previous update step:

```cpp
  math::Quaternion delta = this->orientation.GetInverse() * rot;
  this->orientation = rot;
```

`delta` is a quaternion which represents the orientation difference between two subsequent update steps. The rate vector is basically `delta`'s equivalent [rotation vector](https://en.wikipedia.org/wiki/Axis%E2%80%93angle_representation) (or the vector part of the [quaternion logarithm](https://en.wikipedia.org/wiki/Quaternion#Exponential,_logarithm,_and_power)), divided by the time difference `dt` (first order approximation). `delta` is already expressed in body frame because of the order of the two operands and the result does not depend on the constant rotation offset  applied in [line 251](https://github.com/tu-darmstadt-ros-pkg/hector_gazebo/blob/f9f6e4362aebeaaf74d8c502f59fbab7ee6ebf23/hector_gazebo_plugins/src/gazebo_ros_imu.cpp#L251).

The left-multiplication of `this->offset_.rot.GetInverse()` was added in https://github.com/tu-darmstadt-ros-pkg/hector_gazebo/commit/d5fcb4a40756559fb94cfbbfeae2abbc63bc5e34 in order to fix #23.

**Comment by JaFeKl** [comment]
Thank you very much for this! It really helped me in understand the code.
